ts/features/idpay/initiative/details/screens/BeneficiaryDetailsScreen.tsx

```diff
@@ -0,0 +1,389 @@
+    beneficiaryDetails.rankingStartDate,
+    O.fromNullable,
+    O.map(formatDate("DD MMM YYYY")),
+    O.getOrElse(() => "-")
+  );
+
+  const rankingEndDateString = pipe(
+    beneficiaryDetails.rankingEndDate,
+    O.fromNullable,
+    O.map(formatDate("DD MMM YYYY")),
+    O.getOrElse(() => "-")
+  );
+
+  const rewardPercentageString = pipe(
+    beneficiaryDetails.refundRule?.accumulatedAmount?.refundThreshold,
+    O.fromNullable,
+    O.map(percentage => `${percentage}%`),
+    O.getOrElse(() => "-")
+  );
+
+  const lastUpdateString = pipe(
+    beneficiaryDetails.updateDate,
+    O.fromNullable,
+    O.map(formatDate("DD MMMM YYYY, hh:mm")),
+    O.map(dateString =>
+      I18n.t("idpay.initiative.beneficiaryDetails.lastUpdate", { dateString })
+    ),
+    O.toUndefined
+  );
+
+  const tableItems: ReadonlyArray<TableItem> = [
+    {
+      label: I18n.t("idpay.initiative.beneficiaryDetails.summary"),
+      value: [
+        {
+          label: I18n.t("idpay.initiative.beneficiaryDetails.status"),
+          value: statusString
+        },
+        {
+          label: I18n.t("idpay.initiative.beneficiaryDetails.endDate"),
+          value: endDateString
+        },
+        {
+          label: I18n.t("idpay.initiative.beneficiaryDetails.amount"),
+          value: amountString
+        },
+        {
+          label: I18n.t("idpay.initiative.beneficiaryDetails.toBeRefunded"),
+          value: toBeRefundedString
+        },
+        {
+          label: I18n.t("idpay.initiative.beneficiaryDetails.refunded"),
+          value: refundedString
+        }
+      ],
+      footer: lastUpdateString
+    },
+    {
+      label: I18n.t("idpay.initiative.beneficiaryDetails.spendingRules"),
+      value: [
+        {
+          label: I18n.t("idpay.initiative.beneficiaryDetails.spendFrom"),
+          value: rankingStartDateString
+        },
+        {
+          label: I18n.t("idpay.initiative.beneficiaryDetails.spendTo"),
+          value: rankingEndDateString
+        },
+        {
+          label: I18n.t("idpay.initiative.beneficiaryDetails.spendPercentage"),
+          value: rewardPercentageString
+        }
+      ]
+    },
+    {
+      label: I18n.t("idpay.initiative.beneficiaryDetails.enrollmentDetails"),
+      value: [
+        {
+          label: I18n.t("idpay.initiative.beneficiaryDetails.enrollmentDate"),
+          value: "-"
+        },
+        {
+          label: I18n.t("idpay.initiative.beneficiaryDetails.protocolNumber"),
+          value: "-"
+        }
+      ]
+    }
+  ];
+
+  return (
+    <>
+      {ruleInfoBox}
+      <Table items={tableItems} />
+    </>
+  );
+};
+
+type RulesInfoBoxProps = {
+  info: string;
+};
+
+const RulesInfoBox = (props: RulesInfoBoxProps) => {
+  const { info } = props;
+
+  const { bottomSheet, present, dismiss } = useIOBottomSheetModal(
+    <Markdown>{info}</Markdown>,
+    <H3>{I18n.t("idpay.initiative.beneficiaryDetails.infoModal.title")}</H3>,
+    700,
```

Comment:
could it be a good idea to refactor this bottomSheet to have reactive height aswell?

Reply:
In this case the content must be scrollable, so the bottomSheet should have fixed height.